### PR TITLE
Add react-text-mask definitions

### PR DESCRIPTION
**Pull request by guilhermehubner**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/react-text-mask/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// Type definitions for react-text-mask 16.0
+// Project: https://github.com/text-mask/text-mask
+// Definitions by: Guilherme Hübner <https://github.com/guilhermehubner/>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.6
+
+// react-text-mask/index.d.ts
```
Comment (andy-ms) [inline types/react-text-mask/index.d.ts]: Comment isn't really helpful...

File: types/react-text-mask/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    "files": [
+        "index.d.ts"
+    ],
+    "compilerOptions": {
+        "module": "commonjs",
+        "lib": [
+            "es6"
+        ],
+        "noImplicitAny": true,
+        "noImplicitThis": false,
```
Comment (andy-ms) [inline types/react-text-mask/tsconfig.json]: Should be `true` for all new packages, same for `strictFunctionTypes`.

File: types/react-text-mask/tslint.json [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+	"extends": "dtslint/dt.json",
+	"rules": {
```
Comment (andy-ms) [inline types/react-text-mask/tslint.json]: Please enable all of these, let me know if you're having trouble with a particular one.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by guilhermehubner** [comment]
@andy-ms done!

File: types/react-text-mask/tslint.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+		"dt-header": true,
```
Comment (andy-ms) [inline types/react-text-mask/tslint.json]: You can just make this file `{ "extends": "dtslint/dt.json" }` now (those rules are all enabled by default)

Comment (guilhermehubner) [reply types/react-text-mask/tslint.json]: :+1: @andy-ms  done

File: types/react-text-mask/index.d.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+interface conformToMaskResult {
+    conformedValue: string;
+    meta: {
+        someCharsRejected: boolean
+    };
+}
+
+export class MaskedInput extends React.Component<MaskedInputProps, any> {}
```
Comment (andy-ms) [inline types/react-text-mask/index.d.ts]: It doesn't look like this is available as a named export. This should probably be `export default class MaskedInput extends React.Component<MaskedInputProps, any> {}`.

Comment (guilhermehubner) [reply types/react-text-mask/index.d.ts]: @andy-ms  You're right! https://github.com/text-mask/text-mask/blob/master/react/src/reactTextMask.js#L6 :sweat_smile: 

I've fix it :+1:

**Comment by andy-ms** [comment]
See test failures -- you just need to sprinkle the `export` keyword around and it's good to go

**Comment by guilhermehubner** [comment]
@andy-ms gogogo? :grin:

**Comment by andy-ms** [comment]
Thanks!

**Comment by r0b1n** [comment]
@guilhermehubner looks like `showMask` option is missing in the defenition.

**Pull request merged**